### PR TITLE
Add Microsoft.Identity.Client package reference in Monitoring.Sdk

### PR DESCRIPTION
**Pull request by dkurepa**
Fix for https://github.com/dotnet/arcade-services/pull/2153

https://github.com/dotnet/arcade/issues/11852

### CODE REVIEW

**Comment by dkurepa** [comment]
The [error](https://dev.azure.com/dnceng/internal/_build/results?buildId=2078827&view=logs&j=c2e05747-45ee-513b-76dd-1a76a5883b6f&t=3c6b8306-7696-5a6e-47e9-4a7eb2205652) says it can't load the Microsoft.Identiy.Client which is a dependancy for Azure.Identity.
It should be getting pulled in automatically, but for some reason it's not (even tho it's available on our public feed).
This might be the case because we're calling the MonitoringPublish task as a target.
Adding the reference to Microsoft.Identity.Client might help

File: src/Monitoring/Sdk/DeployPublisher.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@
 using Microsoft.Azure.Services.AppAuthentication;
 using Microsoft.Build.Framework;
 using Microsoft.Build.Utilities;
+using Microsoft.Identity.Client;
```
Comment (premun) [inline src/Monitoring/Sdk/DeployPublisher.cs:17]: I think the using won't matter much

```suggestion
```

Comment (andriipatsula) [inline src/Monitoring/Sdk/DeployPublisher.cs:17]: Do we need this using?

**Review by premun — APPROVED** [review]

File: src/Monitoring/Sdk/DeployPublisher.cs (hunk 1 above)
Comment (dkurepa) [reply src/Monitoring/Sdk/DeployPublisher.cs]: Not really, I removed it

**Review by premun — APPROVED** [review]

**Comment by premun** [comment]
@dkurepa please also verify there is an appropriate version for this package somewhere set

**Comment by dkurepa** [comment]
> @dkurepa please also verify there is an appropriate version for this package somewhere set

They're both set in the Directory.Packages.props, with versions that should work with each other

**Comment by riarenas** [comment]
@dkurepa this doesn't seem to have helped: https://dnceng.visualstudio.com/internal/_build/results?buildId=2080128&view=logs&j=c2e05747-45ee-513b-76dd-1a76a5883b6f&t=3c6b8306-7696-5a6e-47e9-4a7eb2205652&l=45

**Comment by dkurepa** [comment]
Yup, I tried running `dotnet build ..Monitoring.ArcadeServices.proj ...` on my machine with this changes and it solved the issue, but not on AzDo. @garath do you have any advice on what I could try?

**Comment by dkurepa** [comment]
Tried after nuking the artifacts folder, it actually works fine with our without the `Microsoft.Identity.Client` reference

**Comment by garath** [comment]
I don't have a real answer yet. My guess is that the Monitoring SDK `Microsoft.DotNet.Helix.Sdk`, which is what the monitoring project uses, isn't fully restored or isn't actually available. Perhaps because of some Central Package Version Management confusion. 

This project is referring the SDK _source_ instead of a package. Maybe the SDK project isn't getting restored properly. If so, a step before this one that builds the SDK project specifically should solve it. Or, try referencing the built, published package instead. 

https://github.com/dotnet/arcade-services/blob/9c2ae5ed7ed8346952ef49e40d7586faf27f0317/src/Monitoring/Monitoring.ArcadeServices/Monitoring.ArcadeServices.proj#L4

Helix-Service also uses the Monitoring SDK, but via a package and not source code. Does it have also have problems if upgraded? 

(By the way, be sure the version of this SDK that helix-service uses gets upgraded once it's all working properly.)

**Pull request merged**